gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/EncounterMapper.java

```diff
@@ -0,0 +1,327 @@
+import org.hl7.v3.RCMRMT030101UK04Participant2;
+import org.hl7.v3.TS;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import lombok.RequiredArgsConstructor;
+import uk.nhs.adaptors.pss.translator.util.DateFormatUtil;
+import uk.nhs.adaptors.pss.translator.util.EhrResourceExtractorUtil;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class EncounterMapper {
+    private static final List<String> INVALID_CODES = List.of("196401000000100", "196391000000103");
+    private static final String ENCOUNTER_META_PROFILE = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Encounter-1";
+    private static final String PRACTITIONER_REFERENCE_PREFIX = "Practitioner/";
+    private static final String LOCATION_REFERENCE_PREFIX = "Location/";
+    private static final String LOCATION_REFERENCE_SUFFIX = "-LOC";
+    private static final String PERFORMER_SYSTEM = "http://hl7.org/fhir/v3/ParticipationType";
+    private static final String PERFORMER_CODE = "PPRF";
+    private static final String PERFORMER_DISPLAY = "primary performer";
+    private static final String RECORDER_SYSTEM = "https://fhir.nhs.uk/STU3/CodeSystem/GPConnect-ParticipantType-1";
+    private static final String RECORDER_CODE = "REC";
+    private static final String RECORDER_DISPLAY = "recorder";
+    private static final String TOPIC_CLASS_CODE = "TOPIC";
+    private static final String CATEGORY_CLASS_CODE = "CATEGORY";
+    private static final String ENCOUNTER_KEY = "encounters";
+    private static final String CONSULTATION_KEY = "consultations";
+    private static final String TOPIC_KEY = "topics";
+    private static final String CATEGORY_KEY = "categories";
+    private static final String IDENTIFIER_SYSTEM = "https://PSSAdaptor/";
+
+    private final CodeableConceptMapper codeableConceptMapper;
+    private final ListMapper listMapper;
+
+    public Map<String, List<? extends DomainResource>> mapEncounters(RCMRMT030101UK04EhrExtract ehrExtract, Patient patient) {
+        List<Encounter> encounterList = new ArrayList<>();
+        List<ListResource> consultationList = new ArrayList<>();
+        List<ListResource> topicList = new ArrayList<>();
+        List<ListResource> categoryList = new ArrayList<>();
+
+        Map<String, List<? extends DomainResource>> map = new HashMap<>();
+
+        List<RCMRMT030101UK04EhrComposition> ehrCompositionList = getEncounterEhrCompositions(ehrExtract);
+
+        ehrCompositionList.forEach(ehrComposition -> {
+            var encounter = mapToEncounter(ehrComposition, patient);
+            var consultation = listMapper.mapToConsultation(ehrExtract, encounter);
+
+            var topicCompoundStatementList = getTopicCompoundStatements(ehrComposition);
+            if (CollectionUtils.isEmpty(topicCompoundStatementList)) {
+                // generate a 'flat' consultation
```

Comment:
I mentioned in another PR if there is a need for comments? I guess its up for debate, although my view for this project in particular would be that they add noise, since they are only ever really for TODOs. Perhaps extract common logic that is commented into a separate private method- eg. generateFlatConsultation?